src/install/dependency.zig

```diff
@@ -303,147 +303,148 @@ pub const Version = struct {
         }
 
         pub fn infer(dependency: string) Tag {
+            // empty string means >= 0.0.0
+            if (dependency.len == 0) return .npm;
             switch (dependency[0]) {
-                // npm package
-                '=', '>', '<', '0'...'9', '^', '*', '|' => return Tag.npm,
-
-                '.' => return Tag.folder,
-
+                // =1
+                // >1.2
+                // >=1.2.3
+                // <1
+                // <=1.2
+                // ^1.2.3
+                // *
+                '=', '>', '<', '^', '*' => return .npm,
```

Comment:
This needs to be `|` in the case where someone does something like

```zig
|| 6.0.0
```

Reply:
which believe it or not i think i saw once

Reply:
good thing it's undocumented :smirk:

Reply:
Done in https://github.com/oven-sh/bun/pull/1840/commits/fa5b80869b58401e7850cec15feb7e3c4ea2d2c8